crates/core/src/event/extend.rs

```diff
@@ -111,13 +111,21 @@ impl ExtendEvent for Info {
 }
 
 /// Extend an [`Event`] with `is_valid_masp_tx` data.
-pub struct ValidMaspTx(pub usize);
+pub struct ValidMaspTx(pub (usize, Option<Hash>));
 
 impl ExtendEvent for ValidMaspTx {
     #[inline]
     fn extend_event(self, event: &mut Event) {
-        let Self(masp_tx_index) = self;
-        event["is_valid_masp_tx"] = masp_tx_index.to_string();
+        let Self((masp_tx_index, cmt_hash)) = self;
+        let attribute = if let Some(hash) = cmt_hash {
+            format!("cmt/{hash}/is_valid_masp_tx")
+        } else {
+            "is_valid_masp_tx".to_string()
+        };
+        // TODO: should we put the index of the tx in the block at the root
+        // level of the even attributes instead of as the value of the masp
+        // attribute?
```

Comment:
@sug0

Reply:
More on this: we need the tx index in the client when fetching masp transactions but if we need the index for other things we can think about extracting it to another attribute of the event

Reply:
I think it's better to emit two separate attributes in the same event, like marco was suggesting above.

for instance:
```
event["is_valid_masp_tx"] = $tx_index
event["commitment-hash"] = hash($tx_commitments)
```

Reply:
Just a note, a batch could contain more than one masp tx so we need to index that at the inner tx level. At the moment I'm using a custom attribute string which contains a reference to the specific inner tx like this:

`event["cmt/$hash/is_valid_masp_tx"] = $tx_index`

So, in this case we end up with a duplicated transaction index (which is the same for the entire batch).
I believe this is related to https://github.com/anoma/namada/pull/3103#discussion_r1590999811

Reply:
one of the attributes refers to the index in the block (`is_valid_masp_tx`), the other, refers to one of the txs in the batch (`commitment-hash`); this way, we identify the masp tx by the batch tx index in the block, and then by the index in the batch itself. I think it suffices

Reply:
Ah you want to use the `commitment-hash` attribute in conjunction with the `is_valid_masp_tx` one. We still have one problem cause more than one inner tx in the batch could be a valid masp transaction and this solution would not support that case. We should probably encode a vector of hashes as the value of the `commitment-hash` attribute

Reply:
I see yeah, that makes sense!

Reply:
since we're changing these events anyway, could we also rename the attribute `is_valid_masp_tx`? the name implies it's a boolean, but it's a tx index. something like `masp-tx-block-index`. as for the commitments, something like `masp-tx-batch-indices`, or `masp-tx-batch-commitments`.